### ISSUE TITLE
Resize media on upload 

### ISSUE THREAD

**Issue by abeorch** [LemmyNet/jerboa#1593]
### Pre-Flight checklist

- [X] Did you check to see if this issue already exists?
- [X] This is a single feature request. (Do not put multiple feature requests in one issue)
- [X] This is not a question or discussion. (Use https://lemmy.ml/c/jerboa for that)

### Describe The Feature Request Below

When a user selects a media file (image, video etc. ) that is larger than the permitted size the system parses the object and if it determines it is an image (gif, jog etc) then the system transformations askes if the user would like to reduce the objects size. If the user responds  yes then the system transforms the media into a format within the maximum permitted size and allows it to be posted.

**Comment by dessalines** [comment]
I don't think this is desirable, as we'd need to bundle an entire image suite into jerboa. Better to leave that to other dedicated apps made for the purpose of image compression.

**Comment by abeorch** [comment]
Thinking beyond the user experience into how it could be implemented I guess it could either be done with an optional call to an external image suite (I found https://f-droid.org/packages/ru.tech.imageresizershrinker/ as an example ) or via an API externally.

Perhaps being able to configure a call to an App of this kind in settings would be useful. 

 From a user experience being able to have the transfromation automatically parsed would make things alot eaiser

**Comment by dessalines** [comment]
What I do, is open up my image with a compression app, then click share, and share it to jerboa. I don't think there's any other clean way to do it. There's no way to override the image picker.
